src/picterra/client.py

```diff
@@ -888,7 +901,8 @@ def set_annotations(
             self._api_url(
                 "detectors/%s/training_rasters/%s/%s/upload/bulk/%s/commit/"
                 % (detector_id, raster_id, annotation_type, upload_id)
-            )
+            ),
+            json={"class_id": class_id}
```

Comment:
If I try
```python
client.set_annotations(
    DETECTOR_ID,
    RASTER_ID,
    "outline",
    load_annotations("outline2.geojson"),
    #class_id=get_class_id("class1"),
)
```
which is legit, I get` ["class_id: This field may not be null."` from the API, since we have

```python
class AnnotationUploadCommitSerializer(serializers.Serializer):
    class_id = serializers.UUIDField(required=False)
```
So I guess we need to send the JSON only if _class_id_ is not null

Reply:
Good catch! Fixed and added a test (basically relying on `strict_match` in the "without class_id case").

Reply:
thx, just tested and works great! LGTM